hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/metrics/FlinkStreamWriteMetrics.java

```diff
@@ -0,0 +1,170 @@
+import org.apache.flink.metrics.Histogram;
+import org.apache.flink.metrics.Meter;
+import org.apache.flink.metrics.MetricGroup;
+
+/**
+ * Metrics for flink stream write (including append write, normal/bucket stream write etc.).
+ * Used in subclasses of {@link AbstractStreamWriteFunction}.
+ */
+public class FlinkStreamWriteMetrics extends HoodieFlinkMetrics {
+  private static final String DATA_FLUSH_KEY = "data_flush";
+  private static final String FILE_FLUSH_KEY = "file_flush";
+  private static final String HANDLE_CREATION_KEY = "handle_creation";
+
+  /**
+   * Flush data costs during checkpoint.
+   */
+  private long dataFlushCosts;
+
+  /**
+   * Number of records written in during a checkpoint window.
+   */
+  protected long writtenRecords;
+
+  /**
+   * Current write buffer size in StreamWriteFunction.
+   */
+  private long writeBufferedSize;
+
+  /**
+   * Total costs for closing write handles during a checkpoint window.
+   */
+  private long fileFlushTotalCosts;
+
+  /**
+   * Number of handles opened during a checkpoint window. Increased with partition number/bucket number etc.
+   */
+  private long numOfOpenHandle;
+
+  /**
+   * Number of files written during a checkpoint window.
+   */
+  private long numOfFilesWritten;
+
+  /**
+   * Number of records written per seconds.
+   */
+  protected final Meter recordWrittenPerSecond;
+
+  /**
+   * Number of write handle switches per seconds.
+   */
+  private final Meter handleSwitchPerSecond;
+
+  /**
+   * Cost of write handle creation.
+   */
+  private final Histogram handleCreationCosts;
+
+  /**
+   * Cost of a file flush.
+   */
+  private final Histogram fileFlushCost;
+
+  public FlinkStreamWriteMetrics(MetricGroup metricGroup) {
+    super(metricGroup);
+    this.recordWrittenPerSecond = new DropwizardMeterWrapper(new com.codahale.metrics.Meter());
+    this.handleSwitchPerSecond = new DropwizardMeterWrapper(new com.codahale.metrics.Meter());
+    this.handleCreationCosts = new DropwizardHistogramWrapper(new com.codahale.metrics.Histogram(new SlidingWindowReservoir(100)));
+    this.fileFlushCost = new DropwizardHistogramWrapper(new com.codahale.metrics.Histogram(new SlidingWindowReservoir(100)));
+  }
+
+  @Override
+  public void registerMetrics() {
+    metricGroup.meter("recordWrittenPerSecond", recordWrittenPerSecond);
+    metricGroup.gauge("currentCommitWrittenRecords", () -> writtenRecords);
+    metricGroup.gauge("dataFlushCosts", () -> dataFlushCosts);
+    metricGroup.gauge("writeBufferedSize", () -> writeBufferedSize);
+
+    metricGroup.gauge("fileFlushTotalCosts", () -> fileFlushTotalCosts);
+    metricGroup.gauge("numOfFilesWritten", () -> numOfFilesWritten);
+    metricGroup.gauge("numOfOpenHandle", () -> numOfOpenHandle);
+
+    metricGroup.meter("handleSwitchPerSecond", handleSwitchPerSecond);
+
+    metricGroup.histogram("handleCreationCosts", handleCreationCosts);
+    metricGroup.histogram("handleCloseCosts", fileFlushCost);
+  }
```

Comment:
handleCloseCosts -> fileFlushCosts?

Reply:
Sure, also checked the javadoc. Thanks for point this out~